cmake-modules/az_vcpkg.cmake

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+set(AZ_ROOT_DIR "${CMAKE_CURRENT_LIST_DIR}/..")
+
+macro(az_vcpkg_integrate)
+  # VCPKG Integration
+  if(DEFINED ENV{VCPKG_ROOT} AND NOT DEFINED CMAKE_TOOLCHAIN_FILE)
+    set(CMAKE_TOOLCHAIN_FILE "$ENV{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake"
+        CACHE STRING "")
+  elseif(DEFINED ENV{VCPKG_INSTALLATION_ROOT} AND NOT DEFINED CMAKE_TOOLCHAIN_FILE)
+    set(CMAKE_TOOLCHAIN_FILE "$ENV{VCPKG_INSTALLATION_ROOT}/scripts/buildsystems/vcpkg.cmake"
+        CACHE STRING "")
+  endif()
+  if(DEFINED ENV{VCPKG_DEFAULT_TRIPLET} AND NOT DEFINED VCPKG_TARGET_TRIPLET)
+    set(VCPKG_TARGET_TRIPLET "$ENV{VCPKG_DEFAULT_TRIPLET}" CACHE STRING "")
+  endif()
+endmacro()
+
+macro(az_vcpkg_export)
```

Comment:
I would either use a parameter for the macro like `macro(az_vcpkg_export TARGET_NAME)` or a function with arguments instead of relying on the ${TARGET_NAME} to be set before calling this macro.

Reply:
We still need project name and version to be set as some variable, so that these commands work:
```
configure_file("${CMAKE_CURRENT_SOURCE_DIR}/vcpkg/CONTROL" "${CMAKE_BINARY_DIR}/vcpkg/ports/${TARGET_NAME}-cpp/CONTROL" @ONLY)
configure_file("${CMAKE_CURRENT_SOURCE_DIR}/vcpkg/portfile.cmake" "${CMAKE_BINARY_DIR}/vcpkg/ports/${TARGET_NAME}-cpp/portfile.cmake" @ONLY)
```

Reply:
This is not only about the ${TARGET_NAME}, but see sdk/core/azure-core/vcpkg/Config.cmake.in:
```
find_dependency(nlohmann_json @NLOHMANN_JSON_MIN_REQUIRED_VERSION@)
if(@BUILD_TRANSPORT_CURL@)
  find_dependency(CURL @CURL_MIN_REQUIRED_VERSION@)
endif()
```

We just can't pass all the possible variables as parameters.

Reply:
We need to pass the required variables as parameters.
The targetName is required because if it is not set before calling this macro, the macro will still run using an empty string as targetName. Or even worst, it could take the last value set to targetName which can be not the one we need, but a previously used.

For any variable that starts with `CMake...` we don't need to pass it as parameter because we know CMake will take care of assigning those values.

My approach is, if you are using a global constant variable, then it's fine (like the CMake vairables or maybe the AZ_ROOT_DIR) because they are set once and won't change.
But if you are using a variable that is expected to be different everytime this macro is called, then we need it as parameter and not as a global variable